plugin.js

```diff
@@ -8,6 +8,20 @@ const { Signer, sign, unsign } = require('./signer')
 const kReplySetCookies = Symbol('fastify.reply.setCookies')
 const kReplySetCookiesHookRan = Symbol('fastify.reply.setCookiesHookRan')
 
+function fastifyCookieGetCookie (name) {
+  if (!this[kReplySetCookies]?.size) return null
+
+  const cookies = []
+
+  for (const [cookieName, cookie] of this[kReplySetCookies]) {
+    if (cookieName.split(';', 1)[0] === name) {
+      cookies.push(cookie)
+    }
+  }
+
+  return cookies.length ? cookies : null
```

Comment:
If size is 0, we should get null anyway?

Reply:
Oh. We can set multiple cookies? Thought we disalllow this?

Reply:
During the same request, we might set a cookie with the same name for the domains `blog.example.com` and `example.com`, or for the paths `/foo` and `/bar`

So we do allow multiple cookies that have the same name if their `path` or `domain` attributes are different

getCookie doesn't do an advanced search, so it just checks if the names match, and returns the found cookies

Reply:
If this method is the way forward, it should always return an array. The client can determine if cookies exists via `foundCookies.length`.